# RAG evaluation

Run the current end-to-end RAG quality evaluation against the golden dataset. It evaluates the configured RAG pipeline without the optional LLM judge and does not compare prompts or models.

This makes one generated answer per golden case and reports retrieval, citation, refusal, and latency metrics. Ensure the agent OpenAI settings are available in the environment or `.env` file.

In [1]:
from pathlib import Path
from subprocess import run

project_root = Path.cwd().resolve()
if not (project_root / 'pyproject.toml').is_file():
    project_root = project_root.parents[1]

run(['make', 'evaluate-rag'], cwd=project_root, check=True)

{
  "dataset": "evaluation/datasets",
  "corpus": "fixtures/corpus",
  "mode": "vector",
  "k": 5,
  "metrics": {
    "cases": 100,
    "source_precision": 0.98,
    "source_recall": 0.98,
    "section_precision": 0.8617,
    "section_recall": 0.95,
    "citation_validity": 0.99,
    "citation_coverage": 0.98,
    "refusal_rate": 0.0,
    "refusal_accuracy": 1.0,
    "error_rate": 0.0,
    "mean_retrieval_latency_ms": 38.31,
    "mean_generation_latency_ms": 14535.98,
    "mean_total_latency_ms": 14574.58
  }
}


**Conclusion:** The current vector RAG pipeline is operationally reliable on this 100-case dataset, with no errors, 98% source precision/recall, and 99% citation validity. Section recall remains high at 95%, but 86.2% section precision indicates that some answers cite extra or imprecise sections. Retrieval is fast (38 ms); answer generation is the dominant cost at about 14.5 seconds per case before asynchronous overlap. Refusal accuracy cannot be assessed from this run because the dataset contains no unanswerable cases.